PizzaShopOnline.BAU.Site/Controllers/DeliveryController.cs

```diff
@@ -1,27 +1,50 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using PizzaShopOnline.BAU.Site.Models;
+using PizzaShopOnline.BAU.Site.Repositories;
+using PizzaShopOnline.BAU.Site.Services;
+using PizzaShopOnline.BAU.Site.ViewModels;
 
 namespace PizzaShopOnline.BAU.Site.Controllers
 {
     public class DeliveryController : Controller
     {
+        private readonly IPizzaRepository _pizzaRepository;
+
+        public DeliveryController(IPizzaRepository pizzaRepository)
+        {
+            _pizzaRepository = pizzaRepository;
+        }
+
         [HttpGet]
-        public ActionResult DeliveryForm()
+        public ActionResult DeliveryForm( int id)
         {
-            return View();
+              PizzaModel deliveryPizzaModel = _pizzaRepository.GetPizzaModel(id);
```

Comment:
Move all this code from controller to PizzaService model, all of the mapping from model to view model should be done in that service, so then your controller will be thin and only responsible for what it is responsible for.